File: PRACTICA AUXI/python/POLIMORFISMO/Animal.py
"NOMBRE: QUENTA MAMANI NICK RANDY"
"CI: 10037571"

"7. Se hace referencia a algunos animales mediante las siguientes clases:"

"a) Instanciar 1 Perro, 1 Gato y 1 Pájaro."
"b) Sobrecargar el método hacerSonido() para que cada animal emita su sonido característico."

"c) Implementar un método moverse() que indique cómo se mueve cada animal"
"(correr, saltar, volar, etc.)."


class Animal:
    def __init__(self, nombre):
        self.nombre = nombre

    def hacer_sonido(self, sonido=None, accion=None):
        if sonido is None and accion is None:
            print("sonido.")
        elif accion is None:
            print(f"{self.nombre} dice: {sonido}")
        else:
            print(f"{self.nombre} dice: {sonido} mientras está {accion}")


class Perro(Animal):
    def __init__(self, nombre):
        super().__init__(nombre)


class Gato(Animal):
    def __init__(self, nombre):
        super().__init__(nombre)


class Pajaro(Animal):
    def __init__(self, nombre):
        super().__init__(nombre)


if __name__ == "__main__":
    perro = Perro("toto")
    gato = Gato("pata fofa")
    pajaro = Pajaro("kay")

    print("Sonidos de los animales:")
    perro.hacer_sonido("au au", " corriendo")
    gato.hacer_sonido("Miau", "saltando")
    pajaro.hacer_sonido("Pío", "volando")

File: PRACTICA AUXI/java/POLIMORFISMO/Animal.java
// NOMBRE: QUENTA MAMANI NICK RANDY
//CI: 10037571

//7. Se hace referencia a algunos animales mediante las siguientes clases:
//
//a) Instanciar 1 Perro, 1 Gato y 1 Pájaro.
//b) Sobrecargar el método hacerSonido() para que cada animal emita su sonido
//característico.
//
//c) Implementar un método moverse() que indique cómo se mueve cada animal
//(correr, saltar, volar, etc.).

public class Animal {
    private String nombre;

    public Animal(String nombre) {
        this.nombre = nombre;
    }

    public void hacerSonido() {
        System.out.println("sonido.");
    }

    public void hacerSonido(String sonido) {
        System.out.println(nombre + " dice: " + sonido);
    }

    public void hacerSonido(String sonido, String accion) {
        System.out.println(nombre + " dice: " + sonido + " mientras está " + accion + ".");
    }

    public static class Perro extends Animal {
        public Perro(String nombre) {
            super(nombre);
        }
    }

    public static class Gato extends Animal {
        public Gato(String nombre) {
            super(nombre);
        }
    }

    public static class Pajaro extends Animal {
        public Pajaro(String nombre) {
            super(nombre);
        }
    }

    public static void main(String[] args) {

        Perro perro = new Perro("toto");
        Gato gato = new Gato("pata fofa");
        Pajaro pajaro = new Pajaro("jak");

        System.out.println("Sonidos de los animales:");
        perro.hacerSonido("AU AU", "corriendo");
        gato.hacerSonido("Miau", "saltando");
        pajaro.hacerSonido("Pío", "volando");
    }
}
